### ISSUE TITLE
'authorization' event not firing.

### ISSUE THREAD

**Issue by AngSin** [mauron85/react-native-background-geolocation#178]
'authorization' event is fired if location permission is turned off on iOS but not when location permission is turned on again. I use an older version of the plugin, so maybe it was fixed later on but I found no closed issue with such a topic.

## Your Environment

* Plugin version: 0.4.0-rc.2
* Platform: iOS 
* OS version: 11.3
* Device manufacturer and model: iPhone 6
* React Native version: 0.49.0
* Plugin configuration options: BackgroundGeolocation.configure({
     desiredAccuracy: 10,
     stationaryRadius: 1,
     distanceFilter: 1,
     notificationTitle: 'Background tracking',
     notificationText: 'enabled',
     startOnBoot: true,
     stopOnTerminate: true,
     locationProvider: BackgroundGeolocation.DISTANCE_FILTER_PROVIDER,
     interval: 1000,
     fastestInterval: 1000,
     activitiesInterval: 1000,
     stopOnStillActivity: false
    });
* Link to your project: Private

## Context

The authorization event listener should tell the user if the authorization has changed. Moreover, the authorization value provided in the callback function BackgroundGeolocation.checkStatus() function is also different.

## Expected Behavior

An event should be fired on each authorization status change.

## Actual Behavior

'authorization' event is fired if location permission is turned off on iOS but not when location permission is turned on again.
## Steps to Reproduce

1. Create an on.('authorization') event listener that console.logs events fired
2. Turn off iOS location permission
3. Turn on iOS location permission

## Context

I need to know when the user switches location permission both on and off.

**Comment by mauron85** [comment]
Yes please try latest alpha. There were some changes related to authorization. You can also try to read CHANGES.md, maybe answer is there.

**Comment by ntelkedzhiev** [comment]
Any updates on when/whether this will be fixed?
